### ISSUE TITLE
Create a workflow for an internal Hestia user (Charles) to be able to create new bubbles/experience deployments without developer intervention

### ISSUE THREAD

**Issue by alexbfree** [hestiaAI/hestialabs-experiences#829]
### Context 

As a result of https://github.com/hestiaAI/internal-discussions/discussions/19 we decided we want to build functionality to allow bubbles and experience instances to be deployed automatically. We (@andreaskundig @foucault-dumas @alexbfree) realised that a key solution to this is that non-developers (especially customer facing staff) need to be able to pull together experiences and bubble configurations in certain ways. i.e. a particular bubble, containing both a participant experience set and an aggregator experience set (like we did for TL). These created experience pair/bubbles might be used as demos only by Hestia staff, or might be used as "have a go yourself" trial environments for specific customers. All created experiences/bubbles need to be secure and password protected.

### Immediate functional requirements

1. A front end page (not on our public website, and password protected by a general admin page password). On this page:
   - all previously created bubble/experience combinations are listed, along with their access URL, current password and findable/hidden status (see below).
   - The deployment status of these bubbles is shown. they can be deployed to <generated-bubble-name-stub>.hestia.ai (for commercial bubbles) or <generated-bubble-name-stub> (for data collectives bubbles).
   - The live 
3. From this page, the user after entering the general password for this page, can click a button to Create a New Bubble.
4. User chooses a name for the bubble. 
5. User chooses whether the bubble will be deployed to hestia.ai or hestialabs.org URL
6. The user then chooses an aggregator password and a participants' password.
7. The user then is presented with the experiences catalog (#828) and can select which experiences exist in the aggregator view and in the participant view
8. The bubble is created and deployed (and the user has some feedback on the status page of deployment status)
9. The user has the option to remove or add experiences from the bubble and redeploy (if this would result in data loss, please warn the user)
10. The user has the option to delete an experience, which will remove it from the admin page and also un-deploy it. (Again, they are warned about data loss)
11. The user has the option to move an experience from .hestialabs.org to .hestia.ai or vice versa. An experience should only exist in one or other of the two places at one time. (if possible, give user a choice to migrate data or discard. if not easy to do, just discard and warn about data loss (with a cancel option) before proceeding)
12. The user has the option to lock either aggregator/participant experience or both from being live (for example, in the event of misuse, password leak or wanting to block access until payment has been made)

### Technical requirements/suggestions

1. Each bubble/pair of experiences can be either *findable* (which means we can list it in menus on main Hestia site, like reflets.info) or *hidden* (which means you have to have the URL to access it, probably like TL). This is a configurable setting.
2. Bubble name should be unique. Generate URL from bubble name (e.g. convert all to lowercase and convert all non alphanumeric chars to `-` ). If two different names would generate the same URL, the second is also disallowed.
3. a public key for the bubble will need to be generated behind the scenes but the user should not have to worry about this.

### Longer term goals

In the short term, this interface will always be used by Hestia staff (most likely @foucault-dumas). Longer term, we can imagine this being a self service experience, which will introduce additional requirements around paywall, user accounts, locked-out-until-paid or locked-out-once-trial-expired functionality. For now, those matters will be handled outside of this feature and Charles will give access as needed. This longer term approach is to be discussed in a separate discussion ticket (Link to be added), but only once this issue has been deployed and trialled for while.

**Comment by alexbfree** [comment]
@andreaskundig @foucault-dumas please shout if this misses anything we discussed and I will update the above.

**Comment by andreaskundig** [comment]
Has a decision been taken about the domain names? 

It would be simpler to host all our stuff either on hestialabs.org or on hestia.ai, even while we keep the possibility to create a site for a client under a completely different domain.

Similarly, it would be much simpler to have bubbles defined as paths in the url rather than prefixes to the domain: hestialabs.org/my-bubble rather than my-bubble.hestialabs.org, which is much less straightforward technically (there's probably a hacky way to make it work, it could require huge changes in our deployment,  and I'm not entirely sure that it would work without further investigation).

**Comment by pdehaye** [comment]
I think the second (path rather than subdomain) is good. The only advantage I can spot is more granular access control on cookies, but we are currently building things to nullify that gain.

For the first option, I think it is high time to start treating the digipower/education collective as a client ( digipower.academy ). From there one option would be to have the other collectives have their own bubbles.

**Comment by alexbfree** [comment]
@andreaskundig notes this was put on hold.

**Comment by alexbfree** [comment]
It is unclear whether automated bubble creation is still a requirement or whether just having a good github request template is sufficient.

Back to bizdev for that decision.
